File: benchmarks/zig/particles.zig
const std = @import("std");

const N: usize = 100000;

const Particle = struct { x: f64, y: f64, vx: f64, vy: f64 };

var particles: [N]Particle = undefined;

pub fn main(init: std.process.Init) !void {
    var buf: [256]u8 = undefined;
    var w = std.Io.File.stdout().writer(init.io, &buf);
    const out = &w.interface;

    var i: usize = 0;
    while (i < N) : (i += 1) {
        const fi: f64 = @floatFromInt(i);
        particles[i] = .{ .x = fi, .y = fi * 2.0, .vx = 0.5, .vy = -0.5 };
    }

    var step: usize = 0;
    while (step < 100) : (step += 1) {
        i = 0;
        while (i < N) : (i += 1) {
            particles[i].x += particles[i].vx;
            particles[i].y += particles[i].vy;
        }
    }

    try out.print("{d}\n", .{particles[49999].x});
    try out.flush();
}

File: benchmarks/node/particles.js
const N = 100000;
const particles = new Array(N);

for (let i = 0; i < N; i++) {
  particles[i] = { x: i, y: i * 2, vx: 0.5, vy: -0.5 };
}

for (let step = 0; step < 100; step++) {
  for (let i = 0; i < N; i++) {
    const p = particles[i];
    p.x += p.vx;
    p.y += p.vy;
  }
}

console.log(particles[49999].x);
